src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -10799,6 +10799,74 @@ void verify(ModuleSymbol module)
             }
         }
 
+        [Fact, WorkItem(58837, "https://github.com/dotnet/roslyn/issues/58837")]
+        public void GenericAttribute_NestedType()
+        {
+            var source = @"
+using System;
+
+class Container<T>
+{
+    [Attr<T>]
+    class C1
+    {
+    }
+    
+    [AttrContainer<T>.Attr]
+    class C2
+    {
+    }
+    
+    [AttrContainer<T>.B.Attr]
+    class C3
+    {
+    }
+}
+
+class Attr<T> : Attribute { }
+
+class AttrContainer<T>
+{  
+    public class Attr : Attribute
+    {
+        public T Value { get; set; }
+    }
+    
+    public class B
+    {
+        public class Attr : Attribute
+        {
+        }
+    }
+}
+
+[A<(int A, int B)>.B]
+class C
+{
+}
+
+class A<T>
+{  
+    public class B : Attribute
+    {
+    }
+}
```

Comment:
Consider splitting this into a separate test if it is independent of the cases above.

Reply:
f161a6d488f331211b543861d0f85de9098d083e